lib/puppet_forge/lazy_relations.rb

```diff
@@ -11,9 +11,25 @@ module LazyRelations
     # @private
     def self.included(base)
       base.extend(self)
-      base.after_initialize { @_lazy = {} }
     end
 
+    def parent
+      if self.is_a? Class
+        class_name = self.name
+      else
+        class_name = self.class.name
+      end
+
```

Comment:
Got it. I think, it might be easier to understand as something like this...

 class_name.split("::")[-2] 

(yay for negative indexing). This will return nil if [-2] doesn't exist, which is not necessarily better, but you would probably want to throw an error if it's nil at this point.